omnigibson/action_primitives/curobo.py

```diff
@@ -386,54 +505,118 @@ def compute_trajectories(
-        sim_js_names = list(self.robot.joints.keys())
-        cu_js_batch = lazy.curobo.types.state.JointState(
+        cu_joint_state = lazy.curobo.types.state.JointState(
             position=self._tensor_args.to_device(q_pos),
             # TODO: Ideally these should be nonzero, but curobo fails to compute a solution if so
+            # See this note from https://curobo.org/get_started/2b_isaacsim_examples.html
+            # Motion generation only generates motions when the robot is static.
+            # cuRobo has an experimental mode to optimize from non-static states.
+            # You can try this by passing --reactive to motion_gen_reacher.py.
+            # This mode will have lower success than the static mode as now the optimization
+            # has to account for the robot’s current velocity and acceleration.
+            # The weights have also not been tuned for reactive mode.
             velocity=self._tensor_args.to_device(q_vel) * 0.0,
             acceleration=self._tensor_args.to_device(q_eff) * 0.0,
             jerk=self._tensor_args.to_device(q_eff) * 0.0,
-            joint_names=sim_js_names,
-        ).get_ordered_joint_state(self.mg.kinematics.joint_names)
+            joint_names=self.robot_joint_names,
+        )
+
+        # Update the locked joints with the current joint positions
+        self.update_locked_joints(cu_joint_state, emb_sel)
+
+        cu_js_batch = cu_joint_state.get_ordered_joint_state(self.mg[emb_sel].kinematics.joint_names)
 
         # Attach object to robot if requested
         if attached_obj is not None:
-            obj_paths = [geom.prim_path for geom in attached_obj.root_link.collision_meshes.values()]
-            assert len(obj_paths) <= 32, f"Expected obj_paths to be at most 32, got: {len(obj_paths)}"
-            self.mg.attach_objects_to_robot(
-                joint_state=cu_js_batch,
-                object_names=obj_paths,
+            for ee_link_name, obj in attached_obj.items():
+                assert isinstance(obj, RigidPrim), "attached_object should be a RigidPrim object"
+                obj_paths = [geom.prim_path for geom in obj.collision_meshes.values()]
+                assert len(obj_paths) <= 32, f"Expected obj_paths to be at most 32, got: {len(obj_paths)}"
+
+                position, quaternion = self.robot.links[ee_link_name].get_position_orientation()
+                # xyzw to wxyz
+                quaternion = quaternion[[3, 0, 1, 2]]
+                ee_pose = lazy.curobo.types.math.Pose(position=position, quaternion=quaternion).to(self._tensor_args)
+                self.mg[emb_sel].attach_objects_to_robot(
+                    joint_state=cu_js_batch,
+                    object_names=obj_paths,
+                    ee_pose=ee_pose,
+                    link_name=self.robot.curobo_attached_object_link_names[ee_link_name],
+                    scale=0.7 if attached_obj_scale is None else attached_obj_scale[ee_link_name],
```

Comment:
why is the scale 0.7 here? shouldn't default be 1.0?

Reply:
Changed to 0.99.